src/reanimated2/commonTypes.ts

```diff
@@ -36,12 +46,12 @@ export type ShareableSyncDataHolderRef<T> = {
   __hostObjectShareableJSRefSyncDataHolder: T;
 };
 
-export type MapperRegistry = {
+export type MapperRegistry<T> = {
```

Comment:
This shouldn't be bound to one type. In extreme situations outputs could be two shared values of different type parameter.